zebra-consensus/src/primitives/groth16/params.rs

```diff
@@ -55,164 +34,54 @@ pub struct SaplingParameters {
 
 /// Groth16 Zero-Knowledge Proof spend parameters for the Sprout circuit.
 ///
-/// New Sprout outputs were disabled by the Canopy network upgrade.
+/// Adding value to the Sprout pool was disabled by the Canopy network upgrade.
 pub struct SproutParameters {
     pub joinsplit_prepared_verifying_key: groth16::PreparedVerifyingKey<Bls12>,
 }
 
 impl Groth16Parameters {
-    /// Loads the Sprout and Sapling Groth16 parameters, checking the sizes and hashes of the files.
-    ///
-    /// If the parameters are not present, they are automatically downloaded and cached.
+    /// Loads the Sprout and Sapling Groth16 parameters from the `zebrad` binary, and checks that
+    /// the data is valid.
     ///
     /// # Panics
     ///
-    /// - If the parameters were downloaded to a wrong path.
-    /// - After [`PARAMETER_DOWNLOAD_MAX_RETRIES`] failed download retries.
-    /// - If the downloaded or pre-existing parameter files are invalid.
+    /// If the parameter data in the `zebrad` binary is invalid.
     fn new() -> Groth16Parameters {
-        let params_directory = Groth16Parameters::directory();
-        let sapling_spend_path = params_directory.join(zcash_proofs::SAPLING_SPEND_NAME);
-        let sapling_output_path = params_directory.join(zcash_proofs::SAPLING_OUTPUT_NAME);
-        let sprout_path = params_directory.join(zcash_proofs::SPROUT_NAME);
+        tracing::info!("checking and loading Zcash Sapling and Sprout parameters");
 
-        Groth16Parameters::download_sapling_parameters(&sapling_spend_path, &sapling_output_path);
-        Groth16Parameters::download_sprout_parameters(&sprout_path);
+        let (sapling_spend_bytes, sapling_output_bytes) =
+            wagyu_zcash_parameters::load_sapling_parameters();
+        let sprout_vk_bytes = include_bytes!("sprout-groth16.vk");
```

Comment:
The content of this file needs to be manually checked by a reviewer before merging.

See the `zcashd` PR for details:
https://github.com/zcash/zcash/pull/6661/files#r1248159359

Reply:
I confirm the first 1828 bytes of `sprout-groth16.vk` and `sprout-groth16.params` are identical:

```bash
cmp -n 1828 sprout-groth16.vk ~/.zcash-params/sprout-groth16.params
```